controller/controller_manager.go

```diff
@@ -242,27 +242,21 @@ func GetInstanceManagerCPURequirement(ds *datastore.DataStore, imName string) (*
 }
 
 func isControllerResponsibleFor(controllerID string, ds *datastore.DataStore, name, preferredOwnerID, currentOwnerID string) bool {
-	var err error
-	responsible := false
-
-	ownerDown := false
-	if currentOwnerID != "" {
-		ownerDown, err = ds.IsNodeDownOrDeleted(currentOwnerID)
+	isOwnerUnavailable := func(node string) bool {
```

Comment:
I find the original easier to read but upto you, yours has the advantage of being more explicit about the node=="" case. but I find it's self explanatory that node == "" means that the owner is unavailable :)
```
	// we use this approach so that if there is an issue with the data store
	// we don't accidentally transfer ownership
	isOwnerUnavailable := func(node string) bool {
		nodeDown, err := ds.IsNodeDownOrDeleted(node)
		if node != "" && err != nil {
			logrus.Errorf("Error while checking if object %v node %v is down or deleted: %v", name, node, err)
		}
		return node == "" || nodeDown
	}
```

Reply:
Thank you. Fixed